test/distributed/test_data_parallel.py

```diff
@@ -805,6 +805,22 @@ def forward(self, inp):
                 r"nn\.ParameterDict is being used with DataParallel but this"):
             model(input)
 
+    @unittest.skipIf(not TEST_CUDA, "CUDA unavailable")
+    @unittest.skipIf(not TEST_MULTIGPU, "multi-GPU not supported")
+    @skipIfRocm
+    def test_remove_prefix_from_state_dict_if_exists(self):
+        model = torch.nn.Sequential(nn.Linear(32, 32),
+                                    nn.ReLU(),
+                                    nn.Linear(32, 16),
+                                    nn.ReLU())
+        dp_model = nn.DataParallel(deepcopy(model).cuda())
+        try:
+            model_copy = deepcopy(model)
+            model_copy.load_state_dict(dp_model.state_dict(), is_parallel=True)
+            self.assertEqual(model.state_dict().keys(), model_copy.state_dict.keys())
+            self.assertEqual(model.state_dict()._metadata.keys(), model_copy.state_dict._metadata.keys())
+        except Exception as error:
+            print(f"Test failed with {error}.")
```

Comment:
This would actually cause the test to pass since you catch and swallow the exception. I think its fine to not have a try-except clause here and let the exception propagate.